src/main/java/io/github/moulberry/notenoughupdates/profileviewer/DungeonPage.java

```diff
@@ -145,7 +145,10 @@ public void drawPage(int mouseX, int mouseY, float partialTicks) {
 		//Catacombs level thingy
 		{
 			if (levelObjCata == null) {
-				float cataXp = Utils.getElementAsFloat(Utils.getElement(profileInfo, "dungeons.dungeon_types.catacombs.experience"), 0);
+				float cataXp = Utils.getElementAsFloat(Utils.getElement(
```

Comment:
I don't like these `Utils.getElementAsFloat(Utils.getElement` chains. Maybe create a wrapper method "getElementAsFloat" with the getElement code at its core?

Reply:
Alternatively, we can use the default elements type to determine an overload. Orrrrr, we could do the *correct* thing and use gsons object binding feature and avoid all of these complications. (As i have done in the collections api)

Reply:
i think this is resolved in contributor chat